src/coordinate.py

```diff
@@ -1,15 +1,43 @@
 """Coordinate module"""
 from __future__ import annotations
-from typing import Tuple
+from typing import Tuple, Optional, Any
 
 
 class Coordinate(object):
     """Abstract each board position."""
 
-    def __init__(self, coordinate: Tuple[int, int]) -> None:
+    def __init__(self, coordinate: Any, secondCoord: Optional[int]) -> None:
         """Instantiate Coordinate with x and y values."""
-        #TODO: add overload constructor which just takes two ints instead of a tuple?
-        self._coordinate: Tuple[int, int] = coordinate
+        #self._coordinate: Tuple[int, int] = coordinate
+        if isinstance(coordinate, tuple): 
+            self.fromTuple(coordinate)
+        elif isinstance(coordinate, int) and secondCoord is not None:
+            self.fromInts(coordinate, secondCoord)
+        elif isinstance(coordinate, dict):
+            self.fromDict(coordinate)
+        else:
+            raise Exception('cannot initialize coordinate from parameters: ' + coordinate + '; ' + secondCoord)
+
+
+
+    def fromDict(self, coordinate: dict) -> None:
+        """Instantiate Coordinate from object"""
+        if coordinate["x"] is not None and coordinate["y"] is not None:
```

Comment:
I think this check can be more thorough, have you tested what happens when 'x' or 'y' don't exist? Does it fail gracefully? What if we pass in `string` instead of `int`? Or floats?

Orrr maybe we don't need to worry about bad data here, and can trust our inputs. Let me know what you think.